### ISSUE TITLE
esp8266: in non-blocking mode, socket.accept raises ETIMEDOUT instead of EAGAIN

### ISSUE THREAD

**Issue by dpgeorge** [micropython/micropython#2947]
Test code:
```python
import socket
s = socket.socket()
s.bind(socket.getaddrinfo("0.0.0.0", 8000)[0][-1])
s.setblocking(False)
s.listen(1)
s.accept()
```
On CPy and uPy unix the accept raises EAGAIN.   On esp8266 it raises ETIMEDOUT.

Note that if the socket is in a blocking mode with a finite timeout (eg s.settimeout(1)) then ETIMEDOUT is the correct error code to raise (matches CPy's socket.timeout exception for this case).

**Comment by pfalcon** [comment]
Fixed by 50de6d2fab9ad67c6df0d74ce4b8c3704d1de090

**Comment by dpgeorge** [comment]
A test for this was added in 458cbacb8f62ad0f33024360af61fb797530c65b
